subscriber.go

```diff
@@ -676,29 +676,16 @@ func (h *handler) handle(ctx context.Context, nextCid cid.Cid, sel ipld.Node, wr
 	}()
 
 	if wrapSel {
-		// Note this branch is nested under wrapSel because wrapSel adds the
-		// semantics that we stop when we hit the `h.latestSync` node. This is
-		// a special case where we are starting at the stop node so we can just
-		// return.
-		if h.latestSync != nil && h.latestSync.(cidlink.Link).Cid == nextCid {
-			// Nothing to do. We've already synced to this cid because we have
-			// it as h.latestSync.
-			log.Infow("Already synced")
-			return nil
-		}
-
 		sel = ExploreRecursiveWithStopNode(h.subscriber.syncRecLimit, sel, h.latestSync)
-	} else if checkAlreadySynced {
-		// Special case when starting at stop node; selector does not check it.
-		if h.latestSync != nil && h.latestSync.(cidlink.Link).Cid == nextCid {
-			// Nothing to do. We've already synced to this cid because we have
-			// it as h.latestSync.
-			log.Infow("Already synced")
-			return nil
-		}
 	}
 
-	err := syncer.Sync(ctx, nextCid, sel)
+	stopNode, err := getStopNode(sel)
```

Comment:
The `err` is completely discarded.  If is really is an error, then log it.  If it is only used to determine if a stopNode was found, then have `getStopNode` return a bool instead.